modin/utils.py

```diff
@@ -628,6 +630,43 @@ def execute(*objs: Iterable[Any], trigger_hdk_import: bool = False) -> None:
             query_compiler.force_import()
 
 
+@contextlib.contextmanager
+def enable_exp_mode() -> Generator[None, None, None]:  # noqa: MD02
+    """
+    Enable experimental mode.
+
+    Similar to using `export MODIN_EXPERIMENTAL=True`, but allows to enable
+    experimental functionality for a small section of code.
+
+    Returns
+    -------
+    Generator[None, None, None]
+
+    Examples
+    --------
+    >>> import modin.pandas as pd
+    >>> import modin.utils
+    >>> import numpy as np
+    >>> df = pd.DataFrame(np.random.rand(100, 32))
+    >>> with modin.utils.enable_exp_mode():
+    >>>   from modin.experimental.pandas import read_pickle_distributed
+    >>>   df._exp.to_pickle_distributed("transactions_train_new*.pkl")
+    >>>   new_df = read_pickle_distributed("transactions_train_new*.pkl")
+    """
+    from modin.config import IsExperimental
+    from modin.core.execution.dispatching.factories.dispatcher import FactoryDispatcher
+
+    old_value = IsExperimental.get()
+    try:
+        IsExperimental.put(True)
+        FactoryDispatcher._update_factory_if_already_initialized()
```

Comment:
can we hide the factory updating to the `IsExperimental.put()` method? We can then make this context-lib function generic, so it could work with other config variables that we tend to change often (for example, `ExperimentalGroupbyImpl`, `ExperimentalNumpyAPI`, `AsyncReading`, ...):
```python
with modin.utils.set_config_value(IsExperimental, value=True):
    ...
```

Reply:
An interesting idea, it seems possible, but let's check it separately.